src/loaders/coco_load.c

```diff
@@ -275,6 +284,10 @@ static int coco_load(struct module_data *m, HIO_HANDLE *f, const int start)
 			if (event->note)
 				event->note += 12;
 
+			if (hio_error(f)) {
```

Comment:
Nitpick: _maybe_ move this just before the `event->note` handling above?

In any case, this looks better.  One place I'm not sure is whether to check
the read `mod->chn` value being positive around line 190: Many other loaders
do not seem to check, and _maybe_ a bad value would be handled elsewhere
(didn't look close enough, though.)

Reply:
The value read for `mod->chn` is an unsigned byte and is bounded to the valid channels range with `& 0x3f`, which is really tacky but technically works (`XMP_MAX_CHANNELS` is 64). The file format document suggests it should only ever be 4 or 8 but I've seen format documents blatantly lie before...

Reply:
OK.  Leaving fine-tuning and final review to you.